src/users_db.rs

```diff
@@ -356,28 +355,25 @@ impl UsersDb {
                 );
                 try!(stmt.query(&[&id]))
             },
-            ReadFilter::Name(name) => {
+            ReadFilter::Name(ref name) => {
                 stmt = try!(
                     self.connection.prepare("SELECT * FROM users WHERE name=$1")
                 );
                 try!(stmt.query(&[&escape(&name)]))
             },
-            ReadFilter::Email(email) => {
+            ReadFilter::Email(ref email) => {
                 stmt = try!(
                     self.connection.prepare("SELECT * FROM users WHERE email=$1")
                 );
-                try!(stmt.query(&[&escape(&email)]))
+                try!(stmt.query(&[&escape(email)]))
             },
-            ReadFilter::Credentials(user, password) => {
-                let mut md5 = Md5::new();
-                md5.input_str(&escape(&password));
-                let password = md5.result_str();
+            ReadFilter::Credentials(ref user, _) => {
```

Comment:
We should probably get rid of the second argument since we're not using it here.

Or we should use it as `password`, find the user with that `username` and then compare the hashed password with the one provided... thus actually looking up a user by credentials.